.circleci/config.yml

```diff
@@ -159,9 +175,132 @@ jobs:
+      - run:
+          name: 'Test Integration - Run integration tests on modified packages on Mac and Windows version of Chrome and Firefox'
+          command: npm run test:ci:integration
+      # This script parses all the XML files generated by mocha and karma to see if any tests failed/errored out
+      # The reason this script exists is that all tests are "exec"'d and running in a single process instead of a parallel
+      # So if a failure happens it will continue to the next test suite in the queue
+      - run:
+          name: 'Check if any integration tests failed'
+          command: |
+            FAILURES=$(find reports -iname 'webex*.xml' -print0 | xargs -0 sed -n 's/.*failures=\"\([^\"]*\).*/\1/p' | xargs)
+            ERRORS=$(find reports -iname 'webex*.xml' -print0 | xargs -0 sed -n 's/.*errors=\"\([^\"]*\).*/\1/p' | xargs)
+
+            echo Failures $FAILURES
+            echo Errors $ERRORS
+
+            if [[ "$FAILURES" -ne *0* ]] || [[ "$ERRORS" -ne *0* ]]; then
+              echo 'Unit tests failed'
+              exit 1
+            fi
+      - store_test_results:
+          path: reports/
+      - store_artifacts:
+          path: reports/junit
+          destination: junit
+
+  version_bump:
+    executor: main-executor
+    steps:
+      - checkout_and_fetch_tags
+      - restore_node_modules
+      - run:
+          name: 'Generate Version Number'
+          command: echo "export VERSION=$(npm run --silent get-next-version | xargs)" >> $BASH_ENV
+      - run: echo ${VERSION}
+      - run:
+          name: 'Build Packages'
+          command: npm run build
+      - run:
+          name: 'Build UMD script'
+          command: npm run build:script -- --versionNumber=${VERSION}
+      - run:
+          name: 'Build Docs'
+          command: npm run build:docs
+      - run:
+          name: 'Bump all package version'
+          command: npm run tooling -- version set ${VERSION} --all
+      - persist_to_workspace:
+          root: ~/webex-js-sdk
+          paths:
+            - packages/node_modules
+      - run:
+          name: 'Commit bumped version'
+          command: |
+            git add docs/ packages/node_modules/*/package.json packages/node_modules/@ciscospark/*/README.md packages/node_modules/@webex/*/package.json packages/node_modules/webex/umd/*.js
+            git commit -m "chore(release): v${VERSION} [skip ci]"
+      - run:
+          name: 'Push to upstream/master'
+          command: git push upstream HEAD:master"
+      - run:
+          name: 'Tag for release'
+          command: git tag "v${VERSION}"
+      - run:
+          name: 'Push tagged release to upstream/master'
+          command: git push upstream v${VERSION}:v${VERSION}"
+
+  npm_publish:
+    executor: main-executor
+    steps:
+      - checkout_and_fetch_tags
+      - restore_node_modules
+      - restore_built_packages
+      - run:
+          name: 'Check if we need to skip npm publishing [skip npm]'
```

Comment:
@adamweeks @InteractiveTimmy do I need to move this, or also add something similar to version_bump, or combine the two to skip version bumping and publishing?

The current expectation is that merge to master will bump version but then won't publish to npm

Reply:
What does the jenkins pipeline do now with a "no-push"? I would think that we would want to bump the version and publish in one stage/job and make the whole thing skippable if need be (for documentation or other non-code related changes).

Reply:
`no-push` gets hit before version bumping and publishing
I think it's easier to combine them into one job and skip them separate jobs